### PR TITLE
Cleanup xpack build.gradle

### PR DESCRIPTION
**Pull request by jakelandis**
This commit does three things:

* Removes all Copyright/license headers for the build.gradle files under x-pack. (implicit Apache license)
* Removes `evaluationDependsOn(xpackModule('core'))` from build.gradle files under x-pack
* Removes a place holder test in favor of disabling the test task (in the async plugin)

========

This PR looks like alot of changes...but it is really only removing headers, removing an uncessary evaluationDependsOn, and removes a dummy test.

### CODE REVIEW

File: x-pack/plugin/async/build.gradle [hunk 1]
```diff
@@ -25,5 +22,7 @@ dependencyLicenses {
   ignoreSha 'x-pack-core'
 }
 
+//no tests
+test.enabled = false
```
Comment (mark-vieira) [inline x-pack/plugin/async/build.gradle:26]: Can we similarly use the lazy configuration apis here, ie. `tasks.named("test").configure { }`.

File: x-pack/plugin/sql/build.gradle [hunk 2]
```diff
@@ -1,5 +1,13 @@
-evaluationDependsOn(xpackModule('core'))
-
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+/*
```
Comment (mark-vieira) [inline x-pack/plugin/sql/build.gradle:6]: Looks like we snuck in a second header.

**Review by mark-vieira — COMMENTED** [review]
> Removes evaluationDependsOn(xpackModule('core')) from build.gradle files under x-pack
Yay!

I'm not completely convinced that our build files should be licensed under the Elastic license just because they pertain to x-pack projects. @jasontedor do you have an opinion on this?

File: x-pack/plugin/sql/build.gradle (hunk 2 above)
Comment (jakelandis) [reply x-pack/plugin/sql/build.gradle:6]: double the copyright !  good spot.

File: x-pack/plugin/async/build.gradle (hunk 1 above)
Comment (jakelandis) [reply x-pack/plugin/async/build.gradle]: done https://github.com/elastic/elasticsearch/pull/60554/commits/b541ac89d257eca864a0ce89b174e21fa3918612

File: x-pack/plugin/sql/build.gradle (hunk 2 above)
Comment (jakelandis) [reply x-pack/plugin/sql/build.gradle]: decided to remove all headers... so this is now gone.

**Comment by jakelandis** [comment]
> I'm not completely convinced that our build files should be licensed under the Elastic license just because they pertain to x-pack projects. @jasontedor do you have an opinion on this?

Had a side discussion and decided to remove all the headers and let the implicit Apache license take effect. The goal here was not to apply a license, just to make all of the files the same, which this PR now does.

**Review by mark-vieira — APPROVED** [review]

**Pull request merged**